packages/graphback-runtime/src/data/RESTDataProvider.ts

```diff
@@ -0,0 +1,157 @@
+
+/**
+ * RESTDataProvider is providing the required functionality to connect with a RESTful Datasource.
+ * It supports all the standard CRUD operations.
+ * 
+ */
+export class RESTDataProvider<Type = any, GraphbackContext = any> implements GraphbackDataProvider<Type, GraphbackContext>{
+    // Base url of the endpoint. When creating an RESTDataProvider instance, a string is passed into the constructor.
+    protected baseUrl : string;
+    protected baseType : GraphQLObjectType;
+    protected headers : any;
+    protected tableMap: ModelTableMap;
+
+    //constructor
+    public constructor(baseType: GraphQLObjectType, baseUrl:string, headers:any) {
+      this.baseUrl = baseUrl;
+      this.baseType = baseType;
+      this.headers = headers;
+      this.tableMap = buildModelTableMap(baseType);
+    }
+
+    /** 
+     * @param data - JSON object which carries the data needed for the update operation
+     * @param context -passed by the resolvers
+     * 
+     * valid url => (baseUrl+path) 
+     * baseUrl is defined at instance creation via the constructor. path is created from baseType name and other passed values.
+     * 
+     */
+    public async create(data: Type, context?: GraphbackContext): Promise<Type> {
+        const url = `${this.baseUrl}/${this.baseType.name.toLowerCase()}/`
+        const res = await fetch(url, {
+            method: 'post',
+            body:    JSON.stringify(data),
+            headers: this.headers,
+        })
+        const json = await res.json();
+
+        return json;
+    }
+
+
+    /**
+     * 
+     * @param data - JSON object which carries the data needed for the update operation
+     * @param context -passed by the resolvers
+     * 
+     * valid url => (baseUrl+path) 
+     * baseUrl is defined at instance creation via the constructor. path is created from baseType name and other passed values.
+     * 
+     */
+    public async update(data: Type, context?: GraphbackContext): Promise<Type> {
+        const { idField } = getDatabaseArguments(this.tableMap, data);
+        if(idField.value === undefined){
+            throw new Error(`Primary field value for type ${this.baseType} not provided`)
+        }else{
+            // eslint-disable-next-line prefer-template
+            const url = this.baseUrl+`/${this.baseType.name.toLocaleLowerCase()}`+`/${data[idField.name]}`
+            const res = await fetch(url,{
+                method: 'PUT',
+                body:    JSON.stringify(data),
+                headers: this.headers,
+                })
+            const json = await res.json()
+
+            return json;
+        }
+     
+    }
+
+    /**
+     * 
+     * @param data - JSON object which carries the data needed for the update operation
+     * @param context -passed by the resolvers
+     * 
+     * valid url => (baseUrl+path) 
+     * baseUrl is defined at instance creation via the constructor. path is created from baseType name and other passed values.
+     * 
```

Comment:
Very strange formatting.  for comments. I would remove redundant linebreaks and follow the patterns we have in other files.